## 0. Importing Libraries

In [60]:
import pandas as pd
import matplotlib.pyplot as plt
from bokeh.plotting import figure, show
import numpy as np
from math import pi
import random



## 1. Read json file and then load into a data frame

In [2]:
path= r"/Users/varnikaprasad/Downloads/kickstarter_data_full.csv"
initdf= pd.read_csv(path)
# initdf

/var/folders/jc/s8m7s_9n3q3fhyv_61mvy6zw0000gp/T/ipykernel_64178/130372481.py:2: DtypeWarning: Columns (30,31,32,33) have mixed types. Specify dtype option on import or set low_memory=False.
  initdf= pd.read_csv(path)


## 2. Inferred Schema

In [3]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------
# Schema information
initdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20632 entries, 0 to 20631
Data columns (total 69 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   index                        20632 non-null  int64  
 1   Unnamed: 0                   20632 non-null  int64  
 2   id                           20632 non-null  int64  
 3   photo                        20632 non-null  object 
 4   name                         20632 non-null  object 
 5   blurb                        20627 non-null  object 
 6   goal                         20632 non-null  float64
 7   pledged                      20632 non-null  float64
 8   state                        20632 non-null  object 
 9   slug                         20632 non-null  object 
 10  disable_communication        20632 non-null  bool   
 11  country                      20632 non-null  object 
 12  currency                     20632 non-null  object 
 13  currency_symbol 

#### Checking Columns with Null Values

In [4]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

print([col for col in initdf.columns if initdf[col].isnull().any()])

['blurb', 'location', 'category', 'friends', 'is_starred', 'is_backing', 'permissions', 'name_len', 'name_len_clean', 'blurb_len', 'blurb_len_clean']


#### After running the above fucntion, we found out the columns with missing values to handle them.  

#### Columns 'blurb', blurb_len', 'blurb_len_clean', 'location', 'name', 'country', 'is_starred', friends', 'permissions', 'is_backing'  have missing values which are required to be handled. 

## 3. Data Cleaning and Transformations

In [6]:
#Checking rows for which specified column contains null
def nonnullcheck(df, col):
    return df[col][df[col].notna()]

In [7]:
# function to check column values of null values 
def nullcheck(df, col):
    return df[col][df[col].isna()]

#insert the column name for which you wish to check
nullcheck(initdf,'blurb_len_clean') 

1411    NaN
6744    NaN
9239    NaN
11708   NaN
14805   NaN
Name: blurb_len_clean, dtype: float64

In [8]:
#Handling blurb
initdf["blurb"].fillna("Missing blurb", inplace=True)
initdf["blurb_len"].fillna(0, inplace=True)
initdf["blurb_len_clean"].fillna(0, inplace=True)

In [9]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------
initdf[['blurb','blurb_len','blurb_len_clean']]

blurb  blurb_len  \
0      MTS ASL Curriculum Workbook is a reproducible ...       21.0   
1      This kit teaches how to print, correct an ugly...       23.0   
2      Establishing a free, world-class, public libra...       18.0   
3      Goal: Introducing a new word into the English ...       21.0   
4      Zu den Artikeln der DSGVO sind die korrespondi...       18.0   
...                                                  ...        ...   
20627  The easy way to be present for story time anyt...       21.0   
20628  A revolutionary App that makes getting fit as ...       20.0   
20629  We all love fatbooth and Ugly booth, why not s...       27.0   
20630  Production android app for cab drivers and tru...        8.0   
20631  We recently launched version 1 of our unique t...       26.0   

       blurb_len_clean  
0                 16.0  
1                 15.0  
2                 10.0  
3                 13.0  
4                 18.0  
...                ...  
20627             16.0  
20628             15.0  
20629             17.0  
20630              6.0  
20631             17.0  

[20632 rows x 3 columns]

In [10]:
# Checking null rows for location
nullcheck(initdf,'location')

239      NaN
528      NaN
8444     NaN
9285     NaN
9290     NaN
9303     NaN
9318     NaN
9499     NaN
9506     NaN
9549     NaN
9670     NaN
9715     NaN
9763     NaN
9801     NaN
9903     NaN
9980     NaN
9987     NaN
10045    NaN
10100    NaN
10174    NaN
10235    NaN
10299    NaN
10448    NaN
10466    NaN
10503    NaN
10509    NaN
10525    NaN
10595    NaN
10619    NaN
11557    NaN
11598    NaN
11636    NaN
11706    NaN
11732    NaN
12718    NaN
12843    NaN
13680    NaN
14086    NaN
14252    NaN
14254    NaN
14619    NaN
14631    NaN
14971    NaN
15600    NaN
18337    NaN
Name: location, dtype: object

In [11]:
# HANDLING LOCATION NULL VALUES
# Replacing the Null values in location column with "No Location Specified"
initdf["location"].fillna("No location specified", inplace=True)

In [12]:
# Chekcing 'name', 'location' and 'country' for rows which has unspecified location
initdf[['name', 'location', 'country']][initdf['location'] == "No location specified"]


name  \
239    THE FLUTTERBIES: Sponsor A Concert! Produce a ...   
528                "The Squad", Directed by Robyn Griggs   
8444                     Photograph the earth from space   
9285   DIA - Designing Interface Architecture (Phase ...   
9290               MORE JACQUES COUSTEAU, LESS HUCK FINN   
9303              MakerBeam: An Open Source Building Kit   
9318                        Offline Wikipedia iPhone app   
9499                   sustainable flower nationwide CSA   
9506                    The Ficly Server Support Project   
9549                                Family Sleep Project   
9670                       Look at how Kickstarter works   
9715                                 The Art of Counting   
9763   Creating a Site for Promoting Open Source Soft...   
9801                               Ride For World Health   
9903   New, jaw-dropping website for the biggest podc...   
9980                                Gps Trail Conference   
9987   No Scroll Email (Custom Timesaving Email Appli...   
10045                                   ElectricalWhat?!   
10100  Recipe Nom - Easy Recipe storing and bookmarki...   
10174            The Civil War Augmented Reality Project   
10235                           What Is In My Birthchart   
10299  Good Dirt Radio: Climate Solutions for Individ...   
10448                         Cleaning up Hawaii's reefs   
10466                  Beer Universe Android Application   
10503                                     9 to 5 Brow$er   
10509                          iGoozex - Free iPhone app   
10525                                Spreading Geek Love   
10595                                            Prophet   
10619                                           SymfonUs   
11557                                Canceled (Canceled)   
11598             Don't Put Down the Twimote! (Canceled)   
11636                              StreetEats (Canceled)   
11706  TwHistory - Those who forget history are doome...   
11732            GOT YOUR BACK: Mission Haiti (Canceled)   
12718                      Fusion in a Bubblegum Machine   
12843  An Environmental Research Mission To Explore P...   
13680  Fabroot - Making 3D Fabrication Available to E...   
14086                         Paranormal Equipment Drive   
14252  Stand Up & Come Together with The Relentless F...   
14254    Radically Efficient Design for People Transport   
14619         Provisional Patent for TESS Cooling System   
14631       Mass Produce the Bison Oleander Time Machine   
14971  Bridging the Informational Divide through Free...   
15600                 Artificial Intelligence (Canceled)   
18337                Lasersaur: Open Source Laser Cutter   

                    location country  
239    No location specified      US  
528    No location specified      US  
8444   No location specified      US  
9285   No location specified      US  
9290   No location specified      US  
9303   No location specified      US  
9318   No location specified      US  
9499   No location specified      US  
9506   No location specified      US  
9549   No location specified      US  
9670   No location specified      US  
9715   No location specified      US  
9763   No location specified      US  
9801   No location specified      US  
9903   No location specified      US  
9980   No location specified      US  
9987   No location specified      US  
10045  No location specified      US  
10100  No location specified      US  
10174  No location specified      US  
10235  No location specified      US  
10299  No location specified      US  
10448  No location specified      US  
10466  No location specified      US  
10503  No location specified      US  
10509  No location specified      US  
10525  No location specified      US  
10595  No location specified      US  
10619  No location specified      US  
11557  No location specified      US  
11598  No location specified      US  
11636  No location specified      US  
1170

#### Checking null values in columns: 'is_starred', friends', 'permissions' &'is_backing'.

In [13]:
#Checing the Non-null Values and evaluating their impact. 
nonnullcheck(initdf, 'is_backing')

15831    False
15832    False
15833    False
15834    False
15835    False
15836    False
15837    False
15838    False
15839    False
15840    False
15841    False
15842    False
15843    False
15844    False
15845    False
15846    False
15847    False
15848    False
15849    False
15850    False
15851    False
15852    False
15853    False
15854    False
15855    False
15856    False
15857    False
15858    False
15859    False
15860    False
15861    False
15862    False
15863    False
15864    False
15865    False
15866    False
15867    False
15868    False
15869    False
15870    False
15891    False
15892    False
15893    False
15894    False
15895    False
15896    False
15897    False
15898    False
15899    False
15900    False
15901    False
15902    False
15903    False
15904    False
15905    False
15906    False
15907    False
15908    False
15909    False
15910    False
Name: is_backing, dtype: object

In [14]:
#Checing the Non-null Values and evaluating their impact. 
nonnullcheck(initdf, 'is_starred')

15831    False
15832    False
15833    False
15834    False
15835    False
15836    False
15837    False
15838    False
15839    False
15840    False
15841    False
15842    False
15843    False
15844    False
15845    False
15846    False
15847    False
15848    False
15849    False
15850    False
15851    False
15852    False
15853    False
15854    False
15855    False
15856    False
15857    False
15858    False
15859    False
15860    False
15861    False
15862    False
15863    False
15864    False
15865    False
15866    False
15867    False
15868    False
15869    False
15870    False
15891    False
15892    False
15893    False
15894    False
15895    False
15896    False
15897    False
15898    False
15899    False
15900    False
15901    False
15902    False
15903    False
15904    False
15905    False
15906    False
15907    False
15908    False
15909    False
15910    False
Name: is_starred, dtype: object

In [15]:
#Checing the Non-null Values and evaluating their impact. 

nonnullcheck(initdf, 'friends')

15831    []
15832    []
15833    []
15834    []
15835    []
15836    []
15837    []
15838    []
15839    []
15840    []
15841    []
15842    []
15843    []
15844    []
15845    []
15846    []
15847    []
15848    []
15849    []
15850    []
15851    []
15852    []
15853    []
15854    []
15855    []
15856    []
15857    []
15858    []
15859    []
15860    []
15861    []
15862    []
15863    []
15864    []
15865    []
15866    []
15867    []
15868    []
15869    []
15870    []
15891    []
15892    []
15893    []
15894    []
15895    []
15896    []
15897    []
15898    []
15899    []
15900    []
15901    []
15902    []
15903    []
15904    []
15905    []
15906    []
15907    []
15908    []
15909    []
15910    []
Name: friends, dtype: object

In [16]:
#Checing the Non-null Values and evaluating their impact. 

nonnullcheck(initdf, 'permissions')

15831    []
15832    []
15833    []
15834    []
15835    []
15836    []
15837    []
15838    []
15839    []
15840    []
15841    []
15842    []
15843    []
15844    []
15845    []
15846    []
15847    []
15848    []
15849    []
15850    []
15851    []
15852    []
15853    []
15854    []
15855    []
15856    []
15857    []
15858    []
15859    []
15860    []
15861    []
15862    []
15863    []
15864    []
15865    []
15866    []
15867    []
15868    []
15869    []
15870    []
15891    []
15892    []
15893    []
15894    []
15895    []
15896    []
15897    []
15898    []
15899    []
15900    []
15901    []
15902    []
15903    []
15904    []
15905    []
15906    []
15907    []
15908    []
15909    []
15910    []
Name: permissions, dtype: object

### DROPPING UNNECESSARY COLUMNS


In [17]:
# Since the maximum values columns 'friends', 'is_starred', 'is_backing'& 'permissions' is either Null or inserted by the author which will also not affect out visualisations, we will drop these four columns.

initdf.drop(['friends', 'is_starred', 'is_backing', 'permissions'], inplace=True, axis=1)

In [58]:
# To get the look of new Dataframe
initdf.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 20627 entries, 0 to 20631
Data columns (total 65 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   index                        20627 non-null  int64  
 1   Unnamed: 0                   20627 non-null  int64  
 2   id                           20627 non-null  int64  
 3   photo                        20627 non-null  object 
 4   name                         20627 non-null  object 
 5   blurb                        20627 non-null  object 
 6   goal                         20627 non-null  float64
 7   pledged                      20627 non-null  float64
 8   state                        20627 non-null  object 
 9   slug                         20627 non-null  object 
 10  disable_communication        20627 non-null  bool   
 11  country                      20627 non-null  object 
 12  currency                     20627 non-null  object 
 13  currency_symbol 

In [59]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------
nullcheck(initdf, 'category')

Series([], Name: category, dtype: object)

In [50]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

nonnullcheck(initdf, 'category')

0        Academic
1        Academic
2        Academic
3        Academic
4        Academic
           ...   
20627        Apps
20628        Apps
20629        Apps
20630        Apps
20631        Apps
Name: category, Length: 20627, dtype: object

In [51]:
#For all the Null values in categories, we are grouping them and assigning them a value for a cleaner outlay of the dataset.

initdf[['name','category']][initdf['category']=="Uncategorized"]

name       category
120                Ciervos - Electronic Band's 1st Album  Uncategorized
121                            renegadedrum.bandcamp.com  Uncategorized
122                                    Pax's Puppyprints  Uncategorized
123                      Life Is Heller - Me In A Bubble  Uncategorized
124    Let's remix Video Game Music for the Dance Flo...  Uncategorized
...                                                  ...            ...
19579  AreoX Piro Smartphone And GoPro Stabiliser | E...  Uncategorized
19580  Sport X1 stabilizer for action camera & smartp...  Uncategorized
19581  World's most intelligent stabilizer for smartp...  Uncategorized
19582     Omnipod - Any Camera, Every Moment (Suspended)  Uncategorized
19583  Crocodile / To shoot half underwater Go Pro ph...  Uncategorized

[1889 rows x 2 columns]

In [52]:
# Handling categories
initdf["category"].fillna("Uncategorized", inplace=True)

In [53]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

#check if all null values have been handled
print([col for col in initdf.columns if initdf[col].isnull().any()])

[]


In [54]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------


nonnullcheck(initdf, 'name_len')

0        11.0
1         4.0
2         8.0
3         7.0
4         7.0
         ... 
20627     8.0
20628     2.0
20629     3.0
20630     1.0
20631     3.0
Name: name_len, Length: 20627, dtype: float64

In [55]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------


nullcheck(initdf, 'name_len')

Series([], Name: name_len, dtype: float64)

In [61]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------


initdf[['name','name_len','name_len_clean']][initdf['name_len']==0.0]

Empty DataFrame
Columns: [name, name_len, name_len_clean]
Index: []

In [27]:
# DROPPING INCOMPLETE DATA ROWS

initdf.drop(labels=[1411,6744,9239,11708,14805], axis=0, inplace=True)

In [62]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------
# Checking for Null values in Dataframe

initdf.isnull().values.any()

False

## VALIDATION TEST CELLS 

In [29]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

# initdf.info()

In [30]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

# Checking count of categories by their states
# initdf.groupby("state")[["category"]].count()

In [31]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

# Count the number of unique values
# initdf[["category","state"]].nunique()

In [32]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

# List number of unique values
# initdf["state"].unique()

In [33]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

# initdf["category"].unique()

### CREATING DATAFRAMES BY STATE


In [34]:

failed_df = initdf[initdf["state"] == "failed"].groupby(["category"])[['state']].count().reset_index()
successful_df = initdf[initdf["state"] == "successful"].groupby(["category"])[['state']].count().reset_index()
canceled_df = initdf[initdf["state"] == "canceled"].groupby(["category"])[['state']].count().reset_index()
live_df = initdf[initdf["state"] == "live"].groupby(["category"])[['state']].count().reset_index()
suspended_df = initdf[initdf["state"] == "suspended"].groupby(["category"])[['state']].count().reset_index()

### CREATING DEFAULT DICTIONARY FOR STATES

In [35]:


categories_dict = {
    'Academic':0,
    'Places':0,
    'Uncategorized':0,
    'Blues':0,
    'Restaurants':0,
    'Webseries':0, 
    'Thrillers':0, 
    'Shorts':0, 
    'Web':0, 
    'Apps':0, 
    'Gadgets':0,
    'Hardware':0, 
    'Festivals':0, 
    'Plays':0, 
    'Musical':0, 
    'Flight':0, 
    'Spaces':0,
    'Immersive':0, 
    'Experimental':0, 
    'Comedy':0, 
    'Wearables':0, 
    'Sound':0,
    'Software':0, 
    'Robots':0, 
    'Makerspaces':0
}

In [36]:
#    METHOD TO RETURN SERIES PER STATE FOR VISUALIZATION
def populating_state_series(df):

    shallow_copy = categories_dict.copy()
    for i in range(len(df)):
        if df['category'][i] in shallow_copy:
            shallow_copy[df['category'][i]] = df['state'][i]
#     print(shallow_copy)
    return list(shallow_copy.values())

In [37]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------

# categories_dict

In [38]:
# CALLING METHOD TO RETURN SERIES PER STATE FOR VISUALIZATION

categories = list(categories_dict.keys())
failed = populating_state_series(failed_df)
successful = populating_state_series(successful_df)
canceled = populating_state_series(canceled_df)
live = populating_state_series(live_df)
suspended = populating_state_series(suspended_df)

In [39]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------


# print(categories)
# print(successful)
# print(failed)
# print(canceled)
# print(live)
# print(suspended)


In [40]:
# VALIDATION TEST CELL -------------------------------------------------------------------------------------------------


# print(len(canceled), len(successful), len(live), len(suspended), len(failed))
# states = list(initdf["state"].unique())
# print(states)

# 4. VISUALISATIONS

## 4.1 BAR CHART 

In [41]:
from bokeh.plotting import figure, show
from bokeh.plotting import show, output_notebook, output_file
output_notebook()

categories = list(categories_dict.keys())
states = list(initdf["state"].unique())
colors = ["orange", "red", "green","blue", "silver"]

data = {'categories' : categories}
data['failed'] = populating_state_series(failed_df)
data['successful'] = populating_state_series(successful_df) 
data['canceled'] = populating_state_series(canceled_df)
data['live'] = populating_state_series(live_df)
data['suspended'] = populating_state_series(suspended_df)
 

# print(data)
p = figure(x_range=categories, height=500, title="States of Kicstarter by Categories",
           toolbar_location=None, tools='hover', tooltips="$name @categories: @$name")

p.vbar_stack(states, x='categories', width=0.9, color=colors, source=data,
             legend_label=states)

p.y_range.start = 0
p.x_range.range_padding = 0.1
p.xgrid.grid_line_color = None
p.axis.minor_tick_line_color = None
p.outline_line_color = None
p.legend.location = "top_left"
p.legend.orientation = "horizontal"
p.xaxis.major_label_orientation = "vertical"
p.xaxis.axis_label = 'Categories'
p.yaxis.axis_label = 'No.of Kickstarters'


show(p)


Loading BokehJS ...

### 4.1.1 Inference from the Graph

- We have total 25 Categories which contain 20627 kickstarters combined. 



- Top 5 famous categires wuth most kickstarters are:
    1. Web 
    2. Hardware 
    3. Software 
    4. Gadgets 
    5. Uncategorized 
    
    
- Top 5 Successful categories are:
    1. Hardware 
    2. Uncategorized
    3. Plays
    4. Gadgets
    5. Musical
    
    
- Top 5 Failed catergories are:
    1. Web 
    2. Software
    3. Hardware 
    4. Gadgets 
    5. Uncategorized 
  


In [42]:
# METHOD TO RETURN Categories PER STATE FOR VISUALIZATION

def populating_categories_dictionary(df):
    shallow_copy = categories_dict.copy()
    for i in range(len(df)):
        if df['category'][i] in shallow_copy:
            shallow_copy[df['category'][i]] = df['state'][i]
#             shallow_copy['academic'] = 20
#     print(shallow_copy)
    return shallow_copy

## 4.2 PIE CHART

In [57]:
from math import pi
import pandas as pd
from bokeh.io import output_notebook, show
from bokeh.palettes import Category20c
import random
from bokeh.plotting import figure
from bokeh.transform import cumsum
from bokeh.models import LabelSet, ColumnDataSource

output_notebook()

pie_colors=['#039d72','#45BA7E','#de324c', '#f4895f', '#f8e16f',
            '#95cf92','#369acc','#9656a2','#B74E09','#61B22E',
            '#4B2DF7','#5EB999','#5DBDE7','#DD629B','#B2A6A3',
            '#C9212C','#E63DC4', '#A13C50','#4E4327','#76A9CA',
            '#DD7C03','#DD7C03','#80D077','#D84CE4', '#D67956']


total_df = initdf.groupby("category")["state"].count().reset_index()
x = populating_categories_dictionary(total_df)
data = pd.Series(x).reset_index(name='value').rename(columns={'index':'category'})
# print(data)

data['angle'] = data['value']/data['value'].sum() * 2*pi
data['color'] = ['#039d72','#45BA7E','#de324c', '#f4895f', '#f8e16f',
            '#95cf92','#369acc','#9656a2','#B74E09','#61B22E',
            '#4B2DF7','#5EB999','#5DBDE7','#DD629B','#B2A6A3',
            '#C9212C','#E63DC4', '#A13C50','#4E4327','#76A9CA',
            '#DD7C03','#DD7C03','#80D077','#D84CE4', '#D67956']

p = figure(plot_height=800, title="Pie Chart", toolbar_location=None,
           tools="hover", tooltips="@category: @value", x_range=(-0.8, 1.8))

p.wedge(x=0, y=1, radius=0.8,
        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),
        line_color="white", fill_color='color', legend='category', source=data)

data["value"] = data['value'].astype(str)
data["value"] = data["value"].str.pad(35, side = "left")
source = ColumnDataSource(data)

labels = LabelSet(x=0, y=1, text='value',
        angle=cumsum('angle', include_zero=True), source=source, render_mode='canvas')

p.add_layout(labels)

p.axis.axis_label=None
p.axis.visible=False
# p.grid.grid_line_color = None

show(p)

Loading BokehJS ...

### 4.2.1Inference from the Pie Chart


- The above visualisation consist of a simle pie chart showing distribution of all the Kickstarters in various categories



- Pie chart suggests that most of the kickstarters are famous amongts the following categories:
   1. WEB
   2. HARDWARE
   3. SOFTWARE
   4. GADGETS
   5. UNCATEGORIZED

##### CREATING A DATAFRAME FOR LINE CHART

In [63]:
# Using Groupby, Checking count of kickstarter(name) by their launch year which have state as successful. 

initdf[initdf['state'] == 'successful'].groupby('launched_at_yr')['name'].count()

launched_at_yr
2009      15
2010      62
2011     125
2012     262
2013     524
2014    1400
2015    1984
2016    1602
2017      44
Name: name, dtype: int64

In [45]:
# Using Groupby, Checking count of kickstarter(name) by their launch year which have state as Failed. 

initdf[initdf['state'] == 'failed'].groupby('launched_at_yr')['name'].count()

launched_at_yr
2009      26
2010     118
2011     134
2012     227
2013     565
2014    3132
2015    4241
2016    2894
2017      79
Name: name, dtype: int64

In [46]:
# CALLING METHOD TO RETURN SERIES PER Launch Year  FOR VISUALIZATION

# This method will call all the kickstarters according to their launch year.
total_by_yr = initdf.groupby(['launched_at_yr'])[['name']].count().reset_index()

# This method will call all the sucessful kickstarters according to their launch year.
successful_by_yr = initdf[initdf['state'] == 'successful'].groupby('launched_at_yr')['name'].count().reset_index()

# This method will call all the failed kickstarters according to their launch year.
failed_by_yr = initdf[initdf['state'] == 'failed'].groupby('launched_at_yr')['name'].count().reset_index()



In [47]:
# VALIDATION TEST CELL ------------------------------------------------------------------------------------------------
#list(successful_by_yr['name'])

[15, 62, 125, 262, 524, 1400, 1984, 1602, 44]

## 4.3 LINE GRAPH

In [48]:
from bokeh.plotting import figure, show
# prepare some data
x = list(total_by_yr['launched_at_yr']) #list of years
y1 = list(total_by_yr['name']) #count(name/anything) for total by year 
y2 = list(failed_by_yr['name']) #count(name/anything) for failed by year 
y3 = list(successful_by_yr['name']) #count(name/anything) for successful by year 

# create a new plot with a title and axis labels
p = figure(title="Total campaign vs Success/Failure rate", x_axis_label="Year", y_axis_label="Value", plot_width = 600, plot_height = 400)

# add multiple renderers
p.line(x, y1, legend_label="Total.", color="blue", line_width=2)
p.line(x, y2, legend_label="Failed", color="red", line_width=2)
p.line(x, y3, legend_label="Successful", color="green", line_width=2)
# show the results


show(p)

### 4.3.1 Inference from Line Graph

- The graph shows the camparision between the Total campaigns, the total Failed and succesful campaign

- From the graph we can infer that, there are more number of failed campaings than successful ones. 

- Most of the campaigns fromt his dataset is from the year 2013 to year 2018.

- 